File: frontend/src/app/controller/service/admin/entree/InventaireAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import { RoleService } from 'src/app/zynerator/security/Role.service';
import {environment} from 'src/environments/environment';

import {InventaireDto} from 'src/app/controller/model/entree/Inventaire.model';
import {InventaireCriteria} from 'src/app/controller/criteria/entree/InventaireCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class InventaireAdminService extends AbstractService<InventaireDto, InventaireCriteria> {
     constructor(private http: HttpClient, private roleService: RoleService) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/inventaire/');
    }

    public constrcutDto(): InventaireDto {
        return new InventaireDto();
    }

    public constrcutCriteria(): InventaireCriteria {
        return new InventaireCriteria();
    }
}

File: backend-ms1/src/main/java/ma/zs/stocky/service/facade/admin/entree/InventaireAdminService.java
package ma.zs.stocky.service.facade.admin.entree;

import java.util.List;
import ma.zs.stocky.bean.core.entree.Inventaire;
import ma.zs.stocky.dao.criteria.core.entree.InventaireCriteria;
import ma.zs.stocky.zynerator.service.IService;



public interface InventaireAdminService extends  IService<Inventaire,InventaireCriteria>  {

    List<Inventaire> findByCollaborateurId(Long id);
    int deleteByCollaborateurId(Long id);
    long countByCollaborateurPrenom(String prenom);



}
